code/modules/clothing/under/accessories/accessory.dm

```diff
@@ -82,13 +82,17 @@
 		if(istype(U))
 			user.visible_message("<span class='notice'>[user] is putting a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You begin to put a [src.name] on [H]'s [U.name]...</span>")
 			if(do_after(user, 40, target=H) && H.w_uniform == U)
-				user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
-				U.attackby(src, user)
+				if(U.attach_accessory(src, user, TRUE))
+					user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
```

Comment:
Will `[src.name]` add `the` in front of it? If so, might be better to remove the `a` then

Reply:
Nope, but [src] would do that